qt/python/mantidqt/project/workspaceloader.py

```diff
@@ -0,0 +1,27 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2017 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantidqt package
+#
+
+from os import path, listdir
+from mantid.simpleapi import LoadNexusProcessed, LoadMD, Load  # noqa
+from mantid import logger
+
+
+class WorkspaceLoader(object):
+    def load_workspaces(self, directory):
+        filenames = listdir(directory)
+        for filename in filenames:
+            workspace_name, file_ext = path.splitext(filename)
+            if file_ext != u'.project':
+                try:
+                    Load(directory + "/" + filename, OutputWorkspace=workspace_name)
+                except Exception as exception:
+                    if isinstance(exception, KeyboardInterrupt):
```

Comment:
This is unnecessary as `KeyboardInterrupt` inherits `BaseException` and not `Exception` so it will never get caught with this.

Reply:
The idea was to catch all except KeyboardInterrupt errors and fail "Silently"-ish so BaseException wasn't used due to a lack of understanding on my part. :+1:

Reply:
It doesn't look like the exception handling here has been fixed up for the above comment?